CMakeLists.txt

```diff
@@ -13,13 +13,19 @@ if(CMAKE_BUILD_TYPE MATCHES "Release")
 	endif()
 endif()
 
+set(_gnulike GNU Clang)
```

Comment:
This is missing a lot of GNU like compilers (e.g. `AppleClang`). Is there a Modern CMake way to check if a flag is supported maybe?

Reply:
Excellent, at the least, these two options could be implemented with complementary logic--enable them *only* when a known-incompatible compiler (Intel icpc/icl) is being used.

Reply:
in eb77f3fc4 this is done now with [check_cxx_compiler_flag](https://cmake.org/cmake/help/latest/module/CheckCXXCompilerFlag.html)